data/data/bootstrap/files/usr/local/bin/bootstrap-verify-api-server-urls.sh

```diff
@@ -0,0 +1,96 @@
+#!/usr/bin/env bash
+
+# shellcheck disable=SC1091
+. /usr/local/bin/bootstrap-service-record.sh
+
+# This functions expects 2 arguments:
+# 1. name of the URL
+# 2. The value of the URL
+function resolve_url() {
+    unset IPS
+    unset IP
+    IPS=$(dig "${2}" +short)
+    if [[ ! -z "${IPS}" ]] ; then
+        echo "Successfully resolved ${1} ${2}"
+        # dig returns multiple IPs. Check if the
+	# first IP is reachable.
+	ip_arr=""
+	readarray ip_arr -t <<<"${IPS}"
+	IP="$(echo "${ip_arr[0]}" | tr -d '\n')"
+	return 0
+    else
+        echo "Unable to resolve ${1} ${2}"
+	return 1
+    fi
+}
+
+# This functions expects 2 arguments:
+# 1. name of the URL
+# 2. URL to validate
+function validate_url() {
+    if [[ $(curl --head -k --silent --fail --write-out "%{http_code}\\n" "${2}" -o /dev/null) == 200 ]]; then
+        echo "Success while trying to reach ${1}'s https endpoint at ${2}"
+	return 0
+    else
+        echo "Unable to reach ${1}'s https endpoint at ${2}"
+	return 1
+    fi
+
+}
+
+function check_url() {
+    if [[ -z "${1}" ]] || [[ -z "${2}" ]]; then
+	echo "Usage: check_url <API_URL or API_INT URL> <URL that needs to be verified>"
+        return 1
+    fi
+
+    local URL_TYPE=${1}
+    local SERVER_URL=${2}
+
+    if [[ ${URL_TYPE} != API_URL ]] && [[ ${URL_TYPE} != API_INT_URL ]]; then
+	echo "Usage: check_url <API_URL or API_INT URL> <URL that needs to be verified>"
+        return 1
+    fi
+	
+    echo "Checking validity of ${SERVER_URL} of type ${URL_TYPE}"
+
+    if [[ "${URL_TYPE}" = "API_URL" ]]; then
```

Comment:
Based on my other comment, it seems like we don't need to have separate code paths for API_URL vs API_INT_URL, so it seems like this could be simplified

Reply:
For some reason, I am not able to reply directly to comment https://github.com/openshift/installer/pull/5816#discussion_r988271499, so responding here. Using API_INT_CURL_URL="https://${IP}:22623/config/master" as the API-Int URL has 2 advantages: we know that the port 22623 is open and masters will not have an issue getting their configs from the API-Int server. Isn't that the goal of this check?

Reply:
Incorporated by checking for similar URLs for for the API and API-Int servers.